src/eos-autoupdater.c

```diff
@@ -599,10 +601,29 @@ is_time_to_update (guint update_interval_days,
      * computer’s run eos-autoupdater. In order to avoid a thundering herd of
      * computers requesting updates when a lab is first turned on, create a
      * stamp file with a random delay applied, and check again for updates
-     * later. */
-    g_debug ("Not time to update, due to stamp file not being present.");
-    update_stamp_file (update_interval_days, randomized_delay_days);
-    is_time_to_update = FALSE;
+     * later. To do this, we fake the date of the most recent successful update
+     * to be @update_interval_days in the past, so only the
+     * @randomized_delay_days is taken into account for triggering the next
+     * update. */
+    if (randomized_delay_days > 0)
+      {
+        guint64 last_successful_update_secs;
+
+        g_debug ("Not time to update, due to stamp file not being present, but %s is set to %u days.",
+                 RANDOMIZED_DELAY_KEY, randomized_delay_days);
+        last_successful_update_secs = g_get_real_time () / G_USEC_PER_SEC;
+        if (last_successful_update_secs >= update_interval_days * SEC_PER_DAY)
+          last_successful_update_secs -= update_interval_days * SEC_PER_DAY;
```

Comment:
This probably will break in the following scenario (let's assume that the current interval is one week):
- eos-autoupdater runs for the first time, creates a stamp file with mtime set to now minus the interval, so mtime is basically from week ago
- I modify the configuration file, so the interval is two times longer (from one week to two weeks)
- eos-autoupdater runs again, finds out that the next update should be perfomed next week, skip updates

If you deem this scenario to be too strained, then fine, the code looks ok. Otherwise you could consider doing something like:
```
if (stamp_file_does_not_exist)
  {
    if (randomized_delay > 0)
      {
        now = g_get_real_time ();
        first_update_time  = now + randomized_delay_days;
        /* special case - normally the stamp is empty,
         * but we write a byte to let eos-autoupdater
         * know that we should update ASAP + randomized delay
         * without looking at interval days */
        fwrite ("1", 1, 1, stamp_file);
        set_mtime (stamp_file, first_update_time);
        is_time_to_update = FALSE;
      }
    else
      {
        is_time_to_update = TRUE;
      }
  }
else
  {
    if (length_of_stamp_file > 0)
      {
        is_time_to_update = mtime_of_stamp_file < g_get_real_time ();
      }
    else
      {
        is_time_to_update = mtime_of_stamp_file + interval_days < g_get_real_time ();
      }
  }
```

(the pseudocode above blatantly ignores the issues with time units)